client/src/pages/Signup.js

```diff
@@ -95,30 +89,34 @@ const Signup = () => {
     });
   };
 
-  const handleSignup = (e) => {
+  const handleSignup = async (e) => {
     e.preventDefault();
 
     if (formErrors) {
-      setErrorMessage("Please review the form");
+      validateRegistration(dispatch);
       return setSnackbarOpen(true);
     }
 
-    axios
-      .post("/api/users/auth/signup", {
+    try {
+      await register(dispatch, {
         email: signupUser.email,
         name: signupUser.companyName,
         password: signupUser.password,
-      })
-      .then((res) => {
-        setUser(res.data.email, res.data.name);
-        history.push("/");
-      })
-      .catch((err) => {
-        setSnackbarOpen(true);
-        setErrorMessage(err.response.data.message);
       });
+      history.push("/");
+    } catch (err) {
+      setSnackbarOpen(true);
+    }
   };
 
+  useEffect(() => {
+    // is this redirect needed?
+    authenticate(dispatch);
+    if (isAuthenticated) {
+      history.push("/");
+    }
+  }, [isAuthenticated]);
+
```

Comment:
If you're redirecting in the login method I don't think you need it here as well